### PR TITLE
Async image processing

### PR DESCRIPTION
**Pull request by cavis**
- Use `upload_path` and `status` to create pending uploads
- Announce image changes to lambda worker via sns
  - Creates and updates will cause the worker to resize the image, and move to the destination bucket/path.
  - Deletes will hard-delete the mysql record, but the worker will NOT delete the file, instead putting a `deleted.json` file representation of the last state of the record.
- Get callbacks from workers via shoryuken
  - Updates image name/status/size/etc based on the imagemagick identify output

### CODE REVIEW

File: app/controllers/concerns/announce_actions/announce_filter.rb [hunk 1]
```diff
@@ -39,6 +42,7 @@ def announce_resource(action, controller)
     def decorator_class(controller)
       return options[:decorator] if options[:decorator]
       resource_class = controller.controller_name.singularize.camelize
+      "Api::Msg::#{resource_class}Representer".safe_constantize ||
```
Comment (kookster) [inline app/controllers/concerns/announce_actions/announce_filter.rb]: Is this common enough to need to be here, instead of just using the `decorator` option?

File: app/models/audio_file.rb [hunk 2]
```diff
@@ -26,6 +26,7 @@ class AudioFile < BaseModel
   alias_attribute :duration, :length
 
   mount_uploader :file, AudioFileUploader, mount_on: :filename
+  skip_callback :commit, :after, :remove_file! # don't remove s3 file
```
Comment (kookster) [inline app/models/audio_file.rb]: or https://github.com/carrierwaveuploader/carrierwave/wiki/How-to%3a-Keep-uploaded-files-%28for-a-specific-uploader%29

not sure which method is better, the `config.remove_previously_stored_files_after_update = false` seems more directly meant for this.

File: app/controllers/concerns/announce_actions/announce_filter.rb (hunk 1 above)
Comment (cavis) [inline app/controllers/concerns/announce_actions/announce_filter.rb]: Yeah, it's probably going to be just audio and images.  The explicit would probably be better in this case.  Will update!

File: app/models/image.rb [hunk 3]
```diff
@@ -5,9 +5,25 @@ class Image < BaseModel
...
+  NOTFOUND = 'not found'.freeze
+  INVALID  = 'invalid'.freeze
+  FAILED   = 'failed'.freeze
+  COMPLETE = 'complete'.freeze
+
+  alias_attribute :upload, :upload_path
 
   mount_uploader :file, ImageUploader, mount_on: :filename
-  alias_attribute :upload, :remote_file_url
+  skip_callback :commit, :after, :remove_file! # don't remove s3 file
```
Comment (kookster) [inline app/models/image.rb]: same comment as above - and since we generally always want to keep files, I think maybe a global config makes more sense.

File: app/models/audio_file.rb (hunk 2 above)
Comment (cavis) [inline app/models/audio_file.rb]: I think it was [this post](http://stackoverflow.com/questions/17302678/i-would-like-to-stop-carrierwave-deleting-a-file) that led me to believe that only applies to updates.  Will have to double-check that - never actually tried that config.

Comment (cavis) [inline app/models/audio_file.rb]: @kookster we actually already set `config.remove_previously_stored_files_after_update = false` in the carrierwave initializer.

But I confirmed that setting only applies to updates, not deletes.  Without the `skip_callback` the files will all be removed from S3, assuming CMS has permissions to do so.  (If it doesn't, you'll get a 500).

**Comment by cavis** [comment]
@kookster couple more changes to support audio files.

Not exactly sure how to get the "Channel Mode" from ffmpeg output ... but can revisit that later.  For now, it just sets it to single channel or stereo.

Also added some more statuses to `audio_files`.  Does that make sense, or would I be better off sticking to the existing `creating mp3s` and `creating mp3s failed` statuses?

File: app/models/audio_file.rb (hunk 2 above)
Comment (kookster) [inline app/models/audio_file.rb]: my bad, thanks for checking!

**Comment by cavis** [comment]
@kookster ping!

File: app/models/image.rb (hunk 3 above)
Comment (kookster) [inline app/models/image.rb]: and like above, nvm

File: app/representers/api/image_representer.rb [hunk 4]
```diff
@@ -17,19 +17,26 @@ def self_url(represented)
   end
 
   property :id, writeable: false
-  property :filename, writeable: false
+  property :filename
```
Comment (kookster) [inline app/representers/api/image_representer.rb]: I thought carrierwave would prevent direct update of this attribute, why this change?

Comment (cavis) [inline app/representers/api/image_representer.rb]: This was part of my initial pass, where CMS itself (carrierwave) was doing the resize/thumbnails.  Mostly it was useful in dev to leave S3/Lambdas out.

But now that we've moved away from that, I can see removing direct-PUT-ing the file.  Should also update the Dockerfile to not include imagemagick, then.

File: config/environments/development.rb [hunk 5]
```diff
@@ -27,6 +27,9 @@
   # Raise an error on page load if there are pending migrations
   config.active_record.migration_error = :page_load
 
+  # Better logging for docker-compose
+  config.logger = Logger.new(STDOUT)
```
Comment (kookster) [inline config/environments/development.rb]: yeah, I think this should become standard, I was doing something similar on feeder, but this is a better way.

**Comment by kookster** [comment]
whew! 👍
